### PR TITLE
Add Php72 polyfill, abandon Xml polyfill

### PR DESCRIPTION
**Pull request by nicolas-grekas**

### CODE REVIEW

File: src/Php72/README.md [hunk 1]
```diff
@@ -0,0 +1,17 @@
+Symfony Polyfill / Php71
```
Comment (keradus) [inline src/Php72/README.md]: 72

File: src/Php72/Php72.php [hunk 2]
```diff
@@ -9,15 +9,25 @@
...
+    {
+        return function_exists('posix_isatty') && @posix_isatty($stream);
+    }
+
+    public static function sapi_windows_vt100_support()
+    {
+        return false;
+    }
+
     public static function utf8_encode($s)
```
Comment (stof) [inline src/Php72/Php72.php]: this looks weird here. This function is not a function added in PHP 7.2, is it ?

File: src/Php72/README.md [hunk 3]
```diff
@@ -0,0 +1,17 @@
+Symfony Polyfill / Php72
+========================
+
+This component provides functions unavailable in releases prior to PHP 7.2:
+
+- [`stream_isatty`](https://php.net/stream_isatty)
+- [`sapi_windows_vt100_support`](https://php.net/sapi_windows_vt100_support)
+- [`utf8_encode`](https://php.net/utf8_encode)
+- [`utf8_decode`](https://php.net/utf8_decode)
```
Comment (stof) [inline src/Php72/README.md]: this sentence looks weird to me

File: src/Php72/composer.json [hunk 4]
```diff
@@ -0,0 +1,34 @@
...
+        {
+            "name": "Symfony Community",
+            "homepage": "https://symfony.com/contributors"
+        }
+    ],
+    "require": {
+        "php": ">=5.3.3"
+    },
+    "replace": {
+        "symfony/polyfill-xml": "self.version"
```
Comment (stof) [inline src/Php72/composer.json]: you should not have this replacement here, as the metapackage requires this package, which will forbid resolving deps

File: src/Php72/Php72.php (hunk 2 above)
Comment (nicolas-grekas) [inline src/Php72/Php72.php]: yes, see https://github.com/php/php-src/blob/master/NEWS#L227

File: src/Php72/composer.json (hunk 4 above)
Comment (nicolas-grekas) [inline src/Php72/composer.json]: removed

File: src/Php72/README.md (hunk 3 above)
Comment (nicolas-grekas) [inline src/Php72/README.md]: Tweaked. Any suggestion?

Comment (stof) [inline src/Php72/README.md]: Maybe just add a small note saying `(available in the optional XML extension earlier)` ?

Comment (nicolas-grekas) [inline src/Php72/README.md]: updated

File: .travis.yml [hunk 5]
```diff
@@ -9,7 +9,10 @@ cache:
 
 matrix:
     include:
-        - php: hhvm
+        - php: hhvm-3.18
+          sudo: required
+          dist: trusty
+          group: edge
```
Comment (fabpot) [inline .travis.yml]: What about removing HHVM altogether?

File: tests/Php70/Php70Test.php [hunk 6]
```diff
@@ -34,18 +34,24 @@ public function provideIntdiv()
     }
 
     /**
-     * @expectedException ArithmeticError
+     * @expectedException \ArithmeticError
      */
     public function testIntdivArithmetic()
     {
+        if (defined('HHVM_VERSION')) {
+            $this->markTestSkipped('HHVM fails here.');
```
Comment (fabpot) [inline tests/Php70/Php70Test.php]: Removing HHVM really makes sense as just skipping tests that fail for it just means we don't support it anyway.

Comment (nicolas-grekas) [inline tests/Php70/Php70Test.php]: just one of them :)

File: tests/Php70/Php70Test.php [hunk 7]
```diff
@@ -34,18 +34,24 @@ public function provideIntdiv()
...
         intdiv(~PHP_INT_MAX, -1);
     }
 
     /**
-     * @expectedException DivisionByZeroError
+     * @expectedException \DivisionByZeroError
      */
     public function testIntdivByZero()
     {
+        if (defined('HHVM_VERSION')) {
```
Comment (fabpot) [inline tests/Php70/Php70Test.php]: ok, here is another one :)

File: .travis.yml (hunk 5 above)
Comment (fabpot) [inline .travis.yml]: And as we are bumping version to 1.4, it looks like this is a great occasion to do so.

**Comment by fabpot** [comment]
Thank you @nicolas-grekas.

**Comment by keradus** [comment]
awesome !

**Pull request merged**